src/main/java/pedido/comparatorAdicional.java

```diff
@@ -0,0 +1,14 @@
+package pedido;
+
+import ingredientes.Adicional;
+import ingredientes.Ingrediente;
+
+import java.util.Comparator;
+
+public class comparatorAdicional implements Comparator<Adicional> {
+
+    @Override
+    public int compare(Adicional adicional, Adicional t1) {
+        return adicional.obterTipo().toString().compareTo(t1.obterTipo().toString());
```

Comment:
o compare poderia ter sido ajustado nas classes Base, Fruta e Topping da seguinte forma:

    @Override
    public int compareTo(Ingrediente ingrediente) {
        return this.obterTipo().toString().compareToIgnoreCase(ingrediente.obterTipo().toString());
    }

     @Override
     public int compareTo(Ingrediente ingrediente) {
          return this.obterTipo().toString().compareTo(ingrediente.obterTipo().toString());
     }

    @Override
    public int compareTo(Ingrediente ingrediente) {
        return this.obterTipo().toString().compareTo(ingrediente.obterTipo().toString());
    }

creio que assim não precisaria das classes novas com o compare uma vez que as outras implementam Adicional e Ingrediente

Reply:
Para o adicional, dá para fazer apenas a ordenação usando .stream().sorted() e collect no próprio constructor do Shake